cinder/backup/manager.py

```diff
@@ -672,26 +713,42 @@ def restore_backup(self, context, backup, volume_id):
         volume.save()
         backup.status = fields.BackupStatus.AVAILABLE
         backup.save()
-        LOG.info('%(result)s restoring backup %(backup_id)s to volume '
-                 '%(volume_id)s.',
-                 {'result': 'Canceled' if canceled else 'Finished',
-                  'backup_id': backup.id,
-                  'volume_id': volume_id})
+        msg = ('%(result)s restoring backup %(backup_id)s to volume '
+               '%(volume_id)s.' %
+               {'result': 'Canceled' if canceled else 'Finished',
+                'backup_id': backup.id,
+                'volume_id': volume_id})
         self._notify_about_backup_usage(context, backup, "restore.end")
+        trace_utils.trace(
+            context, trace_utils.ACTION_VOLUME_RESTORE,
```

Comment:
Wondering if this action shouldn't be called `ACTION_BACKUP_RESTORE` especially since you put in `backup` as resource.

Reply:
The semantics are odd for sure.  What we are actually doing is restoring a volume from a backup here.  I think that's why I chose the name that I did, as it's not restoring a backup.

Reply:
If it's restoring a volume from a backup, I'd still call it `BACKUP_RESTORE` 🤔 I think everyone's going to read "restoring a backup" as meaning the same as "restoring _from_ a backup", while "restoring a volume" could be misunderstood as "repairing a volume".
But it's a nit.

Reply:
I don't really agree with this, but I'll change it.  it's both a backup restore and a volume restore as a backup is being restored to a volume.   semantics.